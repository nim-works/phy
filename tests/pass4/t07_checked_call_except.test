(TypeDefs
  (Int 8)
  (ProcTy (Void))
)
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 1) (Continuations
    (Continuation (Params) (Locals)
      (Raise (IntVal 100) (Unwind))
    )
  ))
  (ProcDef (Type 1) (Continuations
    (Continuation (Params) (Locals)
      (CheckedCall (Proc 0)
        (Continue 2 (List))
        (Continue 1 (List)))
    )
    (Continuation (Params (Type 0)) (Locals)
      (Continue 2 (List))
    )
    (Continuation (Params))
  ))
)