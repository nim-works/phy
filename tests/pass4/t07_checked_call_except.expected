;$sexp
(TypeDefs
  (Int 8)
  (ProcTy (Void))
)
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 1) (Locals)
    (Continuations
      (Continuation (Params) (Locals)
        (Raise (IntVal 100) (Unwind))
      )
    )
  )
  (ProcDef (Type 1)
    (Locals (Type 0))
    (Continuations
      (Continuation (Params) (Locals)
        (CheckedCall (Proc 0)
          (Continue 2)
          (Continue 1)
        )
      )
      (Except (Local 0)
        (Locals (Local 0))
        (Continue 2)
      )
      (Continuation (Params))
    )
  )
)
